charmcraft/commands/store/client.py

```diff
@@ -126,6 +136,27 @@ def request(self, method, url, body):
         return resp
 
 
+def _storage_push(monitor):
+    """Push bytes to the storage."""
+    session = requests.Session()
```

Comment:
you don't seem to be closing the Session.
Looking over:
https://requests.readthedocs.io/en/master/user/advanced/

it seems to recommend using them as a context manager so you can guarantee their cleanup.

Reply:
Oh, nice! Thanks